openedx/core/djangoapps/enrollments/tests/test_views.py

```diff
@@ -1000,6 +1000,36 @@ def test_update_enrollment_with_expired_mode(self, using_api_key, updated_mode):
         self.assertTrue(is_active)
         self.assertEqual(course_mode, updated_mode)
 
+    @ddt.data(True, False)
+    def test_enrollment_with_global_staff_permissions(self, using_global_staff_user):
+        """Verify no audit enrollments for user different than requesting user and without
+        API_KEY should be done by the users having global staff permissions. """
+
+        CourseModeFactory.create(
+            course_id=self.course.id,
+            mode_slug=CourseMode.VERIFIED,
+            mode_display_name=CourseMode.VERIFIED,
+        )
+
+        if using_global_staff_user:
+            AdminFactory(username='global_staff', email='global_staff@example.com', password=self.PASSWORD)
+            self.client.login(username='global_staff', password=self.PASSWORD)
+        else:
+            self.client.login(username=self.OTHER_USERNAME, password=self.PASSWORD)
+
+        # Create an enrollment
+        self.assert_enrollment_status(
+            as_server=False,
+            mode=CourseMode.VERIFIED,
+            expected_status=status.HTTP_200_OK if using_global_staff_user else status.HTTP_404_NOT_FOUND
```

Comment:
We can pass statuses with `using_global_staff_user` in `ddt.data` instead of if else here:
```
@ddt.data(
    (True, status.HTTP_200_OK),
    (False, status.HTTP_404_NOT_FOUND),
)
def test_enrollment_with_global_staff_permissions(self, using_global_staff_user, expected_status):
```